BuildPropertiesPlugin/README.md

```diff
@@ -0,0 +1,157 @@
+    }
+```
+You can omit the field name and let the plugin generate one for you (following the same rules at 1.)
+
+#### 3. Store whole properties file as generated string resources
+In any product flavor configuration (or `defaultConfig`) you can use `resValueProperties` as follows:
+
+```
+    defaultConfig {
+        ...
+        resValueProperties from(buildProperties.secrets)
+        ...
+    }
+```
+All the entries in that properties files are converted to string resources, named after their key, enforcing snake casing over camel casing if necessary.
+
+
+#### 4. Store a property value as generated string resource
+In any product flavor configuration (or `defaultConfig`) you can use `resValueProperty` as follows:
+
+```
+    defaultConfig {
+        ...
+        resValueProperty 'api_key', from(buildProperties.secrets), 'apiKey'
+        ...
+    }
+```
+You can omit the name of the resource and let the plugin generate one for you from the property key, enforcing snake casing over camel casing if necessary.
+
+
+#### 5. Load signing coinfiguration from properties
+Instead of inline your passwords and other details in your build script you can fill the signing configuration using a properties file.
+```
+signingConfigs {
+  release {
+    signingConfigProperties from(buildProperties.releaseSigning)
+  }
+}
+```
+The plugin will automatically retrieve all the needed fields from the properties file.
+Note: the path of the keystore file is considered relative to the path of the specified properties file.
+
+## Bonus
+
+#### Typed `buildConfigField`/`resValue`
+The plugin enhances the `buildConfigField` and `resValue` facilities to enforce types.
+To generate a string field in your `BuildConfig` you used to write:
+```
+buildConfigField 'String', 'LOL', '\"sometimes the picture takes\"'
+```
+but now you can instead write:
+```
+buildConfigString 'LOL', 'sometimes the picture takes'
+```
+The full list of new typed facilities is as follows:
+
+| | Example |
+|----|----|
+|`buildConfigBoolean` | `buildConfigBoolean 'TEST_BOOLEAN', false`|
+|`buildConfigInt` | `buildConfigInt 'TEST_INT', 42`|
+|`buildConfigLong` | `buildConfigLong 'TEST_LONG', System.currentTimeMillis()`|
+|`buildConfigDouble` | `buildConfigDouble 'TEST_DOUBLE', Math.PI`|
+|`buildConfigString` | `buildConfigString 'TEST_STRING', "whateva"`|
+|`resValueInt`| `resValueInt 'debug_test_int', 100`|
+|`resValueBoolean` | `resValueBoolean 'debug_test_bool', true`|
+|`resValueString` | `resValueString 'debug_test_string', 'dunno bro...'`|
+
+#### More on load properties
+When you need to specify a properties file to retrieve a property from you can use the convenience method `from()` referencing one of the files listed in `buildProperties`. If instead you want to use a different file that is not listed there you can use `load()`, specifying the path of the file you want to use. This in turn will lazy-load the properties file and caching it if any further calls in the build script.
```

Comment:
Not super clear on how `load` differs from `from` after reading this. Maybe an example?